File: Patterns/Abstract Factory Pattern/Cpp/Pizza/Ingredient/ChicagoPizzaIngredientFactory.cpp
#include "./PizzaIngredientFactory.cpp"

#include "./Dough/ThickCrustDough.cpp"
#include "./Cheese/MozzarellaCheese.cpp"
#include "./Sauce/PlumTomatoSauce.cpp"
#include "./Pepperoni/SlicedPepperoni.cpp"
#include "./Clams/FrozenClams.cpp"
#include "./Veggies/Veggies.cpp"
#include "./Veggies/Garlic.cpp"
#include "./Veggies/Onion.cpp"
#include "./Veggies/Mushroom.cpp"
#include "./Veggies/RedPepper.cpp"

class ChicagoPizzaIngredientFactory : public PizzaIngredientFactory {
public:
    Dough* createDough() override {
        return new ThickCrustDough();
    }

    Sauce* createSauce() override {
        return new PlumTomatoSauce();
    }

    Cheese* createCheese() override {
        return new MozzarellaCheese();
    }

    Pepperoni* createPepperoni() override {
        return new SlicedPepperoni();
    }

    Clams* createClams() override {
        return new FrozenClams();
    }

    vector<Veggies*> createVeggies() override {
        return { new Garlic(),new Onion(),new Mushroom(),new RedPepper() };
    }
};

File: Patterns/Abstract Factory Pattern/CSharp/Pizza/Ingredient/ChicagoPizzaIngredientFactory.cs
namespace Pizza.Ingredient
{
    public class ChicagoPizzaIngredientFactory : PizzaIngredientFactory
    {
        public Dough.Dough CreateDough() => new Dough.ThickCrustDough();
        public Sauce.Sauce CreateSauce() => new Sauce.PlumTomatoSauce();
        public Cheese.Cheese CreateCheese() => new Cheese.MozzarellaCheese();
        public Pepperoni.Pepperoni CreatePepperoni() => new Pepperoni.SlicedPepperoni();
        public Clams.Clams CreateClams() => new Clams.FrozenClams();
        public Veggies.Veggies[] CreateVeggies() => 
            new Veggies.Veggies[] { new Veggies.Garlic(),new Veggies.Mushroom(),new Veggies.Onion(),new Veggies.RedPepper() };
    }
}